pkg/hubble/peer/client.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package peer
+
+import (
+	"io"
+
+	peerpb "github.com/cilium/cilium/api/v1/peer"
+)
+
+// Client defines an interface that Peer service client should implement.
+type Client interface {
+	peerpb.PeerClient
+	io.Closer
+}
+
+// ClientBuilder creates a new Client.
+type ClientBuilder interface {
+	// Target returns the address that Client connects to.
+	Target() string
```

Comment:
builder would be `Target(t string) ClientBuilder`, no?

Reply:
maybe I was thinking about another pattern where you chain all of them together `Foo.Target(name).Port(123).Build()`

Reply:
This interface is named `ClientBuilder` because it "builds" a `Client` with the `Client` method. I'm ok to change the name if its confusing but I couldn't come up with a better one at the time.

Reply:
OK, I see your point now. So your suggesting this interface to basically be similar to the `ClientConnBuilder`:
```go
// ClientConnBuilder wraps the ClientConn method.
type ClientConnBuilder interface {
        // ClientConn creates a new ClientConn using target.
        ClientConn(target string) (ClientConn, error)
}
```

The reason I made the `ClientBuilder` interface different is because, as opposed to the `ClientConnBuilder` where each peer address is known by the peers manager, the peer service address is unknown unless another option to provide the peer service target is added. So the `Target()` method was added to the interface mainly to be able to log the peer service address in case of connection problems. At the time, I also thought that having two options to set define the way to connect to the peer service would be error prone. But seeing confusion, I'm also fine updating this interface so the two look similar.

Reply:
`Target()` has been removed from the interface and `Client` now takes `target string` as parameter.